src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final String DEFAULT_COLOR = Color.WHITE.name();
+    private static final double DEFAULT_RADIUS = 10.0;
+    private static final double MAX_RANDOM_VALUE = 10.;
+    private static final int AMOUNT_OF_FIGURES = 4;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        String color = colorSupplier.getRandomColor();
+        int index = random.nextInt(AMOUNT_OF_FIGURES);
+        switch (index) {
+            case 0:
+                return new Square(color, randomSide());
+            case 1:
+                return new Circle(color, randomSide());
+            case 2:
+                return new IsoscelesTrapezoid(color,
+                        randomSide(), randomSide(),
+                        randomSide());
+            case 3:
+                return new RightTriangle(color,
+                        randomSide(),
+                        randomSide());
+            default:
+                return new Rectangle(color,
+                        randomSide(),
+                        randomSide());
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(DEFAULT_COLOR, DEFAULT_RADIUS);
+    }
+
+    private double randomSide() {
+        return random.nextDouble(MAX_RANDOM_VALUE);
```

Comment:
The method `random.nextDouble()` does not take any arguments. To get a double in the range [0, MAX_RANDOM_VALUE), you should multiply the result of `random.nextDouble()` by `MAX_RANDOM_VALUE`.